### PR TITLE
Issue 638

### PR DESCRIPTION
**Pull request by jollopre**
- Sign up email case insensitive.
- Removed session key previous_url since does not work well. 
- Duck-typed User.where_case_insensitive method.

 DMPRoadmap/roadmap#638

### CODE REVIEW

File: app/controllers/registrations_controller.rb [hunk 1]
```diff
@@ -41,18 +41,12 @@ def new
 
   # POST /resource
   def create
-    #logger.debug "#{sign_up_params}"
     if !sign_up_params[:accept_terms] then
       redirect_to after_sign_up_error_path_for(resource), alert: _('You must accept the terms and conditions to register.')
     else
-      existing_user = User.find_by_email(sign_up_params[:email])
-      if !existing_user.nil? # If email exists
-        if (existing_user.password == "" || existing_user.password.nil?) && existing_user.confirmed_at.nil? # If user has not accepted invitation yet
```
Comment (briri) [inline app/controllers/registrations_controller.rb]: Why remove this? It was added to handle situations where a invitation email was sent but the user tries to create an account outside the invitation process. See #322. Devise creates the user record in the DB when it sends out the invitation so the user cannot create an account because the email is already known.

File: app/models/user.rb [hunk 2]
```diff
@@ -285,10 +285,9 @@ def deliver_invitation(options = {})
   ##
   # Case insensitive search over User model
   # @param field [string] The name of the field being queried
-  # @param val [string] The string to search for, case insensitive
+  # @param val [string] The string to search for, case insensitive. val is duck typed to check whether or not downcase method exist
   # @return [ActiveRecord::Relation] The result of the search
   def self.where_case_insensitive(field, val)
-    User.where("lower(#{field}) = ?", val.downcase)
+    User.where("lower(#{field}) = ?", val.respond_to?(:downcase) ? val.downcase : val.to_s)
```
Comment (briri) [inline app/models/user.rb]: a good safety check here. you could also have checked `is_a?(String)` but this is fine too.

Comment (jollopre) [reply app/models/user.rb]: val is always an object with the downcase method check is more than enough here

File: app/controllers/registrations_controller.rb (hunk 1 above)
Comment (jollopre) [reply app/controllers/registrations_controller.rb]: The logic introduced covers more sensibly the edge case for #322. Note before it was just a super, not executing our override lines for when a user does not exist.

Comment (briri) [reply app/controllers/registrations_controller.rb]: ok. will test it out once deployed to make sure #322 hasn't been reintroduced
